src/main/java/com/facebook/openwifirrm/modules/ApiServer.java

```diff
@@ -338,6 +362,26 @@ private void beforeFilter(Request request, Response response) {
 		// Remove "Server: Jetty" header
 		response.header("Server", "");
 
+		// Set CORS headers
+		String origin = request.headers("Origin");
+		if (origin != null) {
+			boolean allowOrigin = false;
+			if (corsDomains == null) {
+				allowOrigin = true;
+			} else {
+				try {
+					String originHost = new URI(origin).getHost();
+					allowOrigin = corsDomains.contains(originHost);
+				} catch (URISyntaxException e) { /* invalid origin URI */ }
```

Comment:
Should there be some kind of log message here?

Reply:
This gets captured in the "CORS checks failed" log below